tools/Vrops.py

```diff
@@ -141,8 +141,25 @@ def get_cluster_and_datastores(self, target, token, parent_uuids):
     def get_hosts(self, target, token, parent_uuids):
         return self.get_resources(target, token, parent_uuids, resourcekinds=["HostSystem"])
 
-    def get_vms(self, target, token, parent_uuids):
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["VirtualMachine"], data_receiving=True)
+    def get_vms(self, target, token, parent_uuids, vcenter_uuid):
+        amount_vms, api_responding, response_time = self.get_latest_stats_multiple(target, token, [vcenter_uuid],
+                                                                                   ['summary|total_number_vms'],
+                                                                                   'Inventory')
+        number_of_vms = amount_vms[0].get('stat-list', {}).get('stat', [])[0].get('data', [0])[0] if \
+            api_responding == 200 else 0
+
+        # vrops cannot handle more than 10000 resource requests
+        if number_of_vms > 10000:
+            uuids_chunked = list(chunk_list(parent_uuids, int(round(len(parent_uuids) / 2, 0))))
```

Comment:
So basically we are chunking every 2 hosts? Am I getting this right?

Reply:
We making two chunks with the half of the hosts each.

Reply:
We assuming a uniform distribution over the hosts, yes.

Reply:
Can we make it two tiers?

```
> 10000 -> 2
> 20000 -> 4
```

Reply:
or better even: every 10000 more we multiply chunks by two.

Reply:
Ok, you are right. I think also we exceeding 20000 VMs in the near future. I adjust this.